src/tools/schema2kotlin.ts

```diff
@@ -234,13 +260,13 @@ export class KotlinGenerator implements ClassGenerator {
   }
 
   // TODO: allow optional fields in kotlin
-  addField({field, typeName, refClassName, isOptional = false, isCollection = false}: AddFieldOptions) {
-    // TODO: support reference types in kotlin
-    if (typeName === 'Reference') return;
+  addField({field, typeName, refClassName, refSchemaHash, isOptional = false, isCollection = false}: AddFieldOptions) {
+    if (typeName === 'Reference' && this.opts.wasm) return;
 
-    const {type, decodeFn, defaultVal} = getTypeInfo(typeName);
+    const {type, decodeFn, defaultVal, schemaType} = getTypeInfo(typeName, refClassName, refSchemaHash);
     const fixed = this.escapeIdentifier(field);
     const quotedFieldName = quote(field);
+    const optionalType = type.endsWith('?') ? type : `${type}?`;
```

Comment:
Why is there an isOptional parameter to addField? Unused? Couldn't we pass optional in Schema2Base and use that instead?

Reply:
Renamed.

I'm not sure exactly why that parameter is there. I think it's probably intended to be for a future feature where we have optional and required fields, but I'm only guessing.